src/node.rs

```diff
@@ -1359,12 +1377,16 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
         inner.filters.notify_new_block(empty_block_hash);
 
         {
-            // That's why here, we increase the batch by 1, but miniblock (and timestamp) by 2.
-            // You can look at insert_fictive_l2_block function in VM to see how this fake block is inserted.
-
             inner.current_batch += 1;
-            inner.current_miniblock += 2;
-            inner.current_timestamp += 2;
+            inner.current_miniblock += 1;
+            inner.current_timestamp += 1;
+
+            // If it was not eth call, we increase the batch by 1, but miniblock (and timestamp) by 2.
+            // You can look at insert_fictive_l2_block function in VM to see how this fake block is inserted.
+            if let TxExecutionMode::VerifyExecute = execution_mode {
```

Comment:
uh, let's avoid this... I'm worried that we'll get bootloader in a bad spot.

Reply:
What exactly? Do you mean a missing fictive block?